pydicom/dataset.py

```diff
@@ -599,6 +607,27 @@ def get_item(self, key):
             return self[key]
         return data_elem
 
+    def _dataset_slice(self, slice):
+        """Return a slice that has the same properties as the original
+        dataset.
+        """
+        tags = self._slice_dataset(slice.start, slice.stop, slice.step)
+        dataset = Dataset({tag: self.get_item(tag) for tag in tags})
+        dataset.read_implicit_vr = self.read_implicit_vr
+        dataset.read_little_endian = self.read_little_endian
+        dataset.is_little_endian = self.is_little_endian
+        dataset.is_implicit_VR = self.is_implicit_VR
+        return dataset
+
+    def write_like_original(self):
+        """Return True if the properties to be used for writing are set and
+        have the same value as the ones in the dataset after reading it.
+        """
+        return (self.is_implicit_VR is not None and
+                self.is_little_endian is not None and
+                self.read_implicit_vr == self.is_implicit_VR and
+                self.read_little_endian == self.is_little_endian)
+
```

Comment:
I'm wondering also about Specific Character Set ... if that had been changed, then data elements should be converted.  It's a very unlikely scenario, but perhaps we should defend against it.

Reply:
Good point! I haven't thought about this.